firebase-vscode/src/analytics.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  env,
+  TelemetryLogger,
+  TelemetrySender,
+} from "vscode";
+import { pluginLogger } from "./logger-wrapper";
+import { AnalyticsParams, trackVSCode } from "./track";
+
+
+export class AnalyticsLogger {
+    readonly logger: TelemetryLogger
+    constructor() {
+        this.logger = env.createTelemetryLogger(new GA4TelemetrySender(pluginLogger));
+    }
+}
+
+class GA4TelemetrySender implements TelemetrySender {
+  constructor(readonly pluginLogger) {}
+
+  sendEventData(
+    eventName: string,
+    data?: Record<string, any> | undefined,
+  ): void {
+    if (!env.isTelemetryEnabled) {
+      pluginLogger.warn("Telemetry is not enabled.")
+    }
+    if (!data) {
+      return;
+    }
+    trackVSCode(eventName, data as AnalyticsParams);
+  }
+
+  sendErrorData(error: Error, data?: Record<string, any> | undefined): void {
```

Comment:
I like the idea here, but be careful what we include in error events when we implement this later. 

Events should not include identifying info like file paths or service ids. If youre relying on error.message to name name the event, its very easy to accidentally include this.